server/sonar-server/src/main/java/org/sonar/server/es/NewIndex.java

```diff
@@ -271,14 +271,15 @@ public Object getProperty(String key) {
   /**
    * Helper to define a string field in mapping of index type
    */
-  public abstract static class StringFieldBuilder {
+  public abstract static class StringFieldBuilder<T extends StringFieldBuilder> {
```

Comment:
I usually use `StringFieldBuilder<T extends StringFieldBuilder<T>>`, when applying this pattern. (Nevertheless that pattern is kind of inconvenient in some situations)

Reply:
:+1: